aineko/extras/connectors/http_poller.py

```diff
@@ -0,0 +1,264 @@
+            f"Creating new session to endpoint {self.http_poller_params.url}."
+        )
+        self.session = requests.Session()
+
+    def _execute(self, params: Optional[Dict] = None) -> None:
+        """Polls and gets data from the HTTP or HTTPS endpoint.
+
+        Raises:
+            Exception: If the retry count exceeds the max retries.
+        """
+        # Check if it is time to poll
+        if (
+            time.time() - self.last_poll_time
+            >= self.http_poller_params.poll_interval
+        ):
+            self.last_poll_time = time.time()
+            raw_data = self.poll_endpoint()
+            if not raw_data:
+                return
+            parsed_data = self.parse_data(raw_data)
+            if not parsed_data:
+                return
+            self.producers[self.output_dataset].produce(parsed_data)
+            self.retry_count = 0
+        else:
+            # If it is not time to poll, sleep
+            time.sleep(
+                self.http_poller_params.poll_interval
+                - (time.time() - self.last_poll_time)
+            )
+
+    def poll_endpoint(self) -> Optional[str]:
+        """Polls the endpoint for data."""
+        try:
+            # Poll the endpoint
+            response = self.session.get(
+                self.http_poller_params.url,
+                timeout=self.http_poller_params.timeout,
+                headers=self.http_poller_params.headers,
+                data=self.http_poller_params.data,
+                params=self.http_poller_params.params,
+                json=self.http_poller_params.json_,
+                auth=self.http_poller_params.auth,
+            )
+            # Check if the request was successful
+            if (
+                response.status_code
+                not in self.http_poller_params.success_codes
+            ):
+                # pylint: disable=broad-exception-raised
+                raise Exception(
+                    f"Request to url {self.http_poller_params.url} "
+                    "failed with status code: "
+                    f"{response.status_code}"
+                )
+            return response.text
+        except Exception as err:  # pylint: disable=broad-except
+            # If request fails, log the error and sleep
+            self.log(
+                "Request failed. "
+                f"Sleeping for {self.http_poller_params.retry_sleep} "
+                f"seconds. Error: {err}",
+                level="error",
+            )
+            time.sleep(self.http_poller_params.retry_sleep)
+            self.retry_count += 1
+            # Reset the session
+            self.log(
+                "Creating new session to HTTP endpoint "
+                f"{self.http_poller_params.url}."
+            )
+            self.session = requests.Session()
+            return None
+
+    def parse_data(self, raw_data: str) -> Optional[Dict[str, Any]]:
+        """Parses raw endpoint response using JSON parser."""
```

Comment:
We should include data on args / raises but I'm OK merging this PR for now given time constraints.

Reply:
same with `poll_endpoint`